### PR TITLE
feat: add trivy-scan-image action

### PR DESCRIPTION
**Pull request by matthewelwell**
Code copy and pasted from [here](https://github.com/Flagsmith/flagsmith/tree/main/.github/actions/trivy-scan-image).

### CODE REVIEW

File: trivy-scan-image/action.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      with:
+        category: ${{ inputs.category }}
+        sarif_file: trivy-results.sarif
+
+    - name: Render scan results URL
+      if: success() || failure()
+      id: scan-results-url
+      run: >
+        echo scan-results-url=${{ format('{0}/{1}/security/code-scanning?query=is:open+tool:Trivy+{2}',
+        github.server_url, github.repository, inputs.query) }} >> $GITHUB_OUTPUT
```
Comment (khvn26) [inline trivy-scan-image/action.yml:76]: Are we sure `github.repository` is going to reference the calling workflow and not the `Flagsmith/actions` one?

**Review by khvn26 — COMMENTED** [review]
This looks ok, apart from one question. Additionally, should we consider migrating the Trivy database [caching cron-triggered workflow](https://github.com/Flagsmith/flagsmith/blob/main/.github/workflows/platform-docker-trivy-scan.yml) over to this repo?

File: trivy-scan-image/action.yml (hunk 1 above)
Comment (matthewelwell) [reply trivy-scan-image/action.yml:76]: Good question. I'll test it out. Not sure why I hadn't already tbh.

Comment (matthewelwell) [reply trivy-scan-image/action.yml:76]: Based on [this workflow run](https://github.com/Flagsmith/flagsmith/actions/runs/11555844042/job/32162204125) in the core repository, and specifically, the following screenshot, I think we're good here. 


image


That being said, I was a little confused by [this](https://github.com/Flagsmith/flagsmith/pull/4766/checks?check_run_id=32162314762) but it does seem like other PRs report similar warnings so I don't think it's related to this PR specifically.

**Review by khvn26 — APPROVED** [review]

**Comment by matthewelwell** [comment]
> Additionally, should we consider migrating the Trivy database [caching cron-triggered workflow](https://github.com/Flagsmith/flagsmith/blob/main/.github/workflows/platform-docker-trivy-scan.yml) over to this repo?

I don't fully understand enough to know if we should or shouldn't move that across. It seems like some of it (the database update) should probably be moved across, but the actual scan functionality probably should stay in the repo that generates the images we are scanning? I'm not sure.

**Comment by matthewelwell** [comment]
I'll go ahead and merge this in the meantime, but let me know your thoughts @khvn26.

**Pull request merged**